### PR TITLE
SPEI

### PR DESCRIPTION
**Pull request by thushari93**
UOC R-Instat collabaration- SPI

### CODE REVIEW

File: instat/SPEI.Designer.vb [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+        Me.RadioButton1.Appearance = System.Windows.Forms.Appearance.Button
+        Me.RadioButton1.AutoSize = True
+        Me.RadioButton1.Location = New System.Drawing.Point(70, 17)
+        Me.RadioButton1.Name = "RadioButton1"
+        Me.RadioButton1.Size = New System.Drawing.Size(63, 23)
+        Me.RadioButton1.TabIndex = 1
+        Me.RadioButton1.TabStop = True
+        Me.RadioButton1.Text = "SPI/SPEI"
+        Me.RadioButton1.UseVisualStyleBackColor = True
+        '
```
Comment (maxwellfundi) [inline instat/SPEI.Designer.vb]: These radio buttons needs to follow the right naming convention. 

eg: rdoXXXXX

File: instat/SPEI.Designer.vb [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+        'GroupBox1
+        '
+        Me.GroupBox1.Controls.Add(Me.RadioButton5)
+        Me.GroupBox1.Controls.Add(Me.RadioButton4)
+        Me.GroupBox1.Location = New System.Drawing.Point(9, 238)
+        Me.GroupBox1.Name = "GroupBox1"
+        Me.GroupBox1.Size = New System.Drawing.Size(149, 52)
+        Me.GroupBox1.TabIndex = 13
+        Me.GroupBox1.TabStop = False
+        Me.GroupBox1.Text = "Index"
```
Comment (maxwellfundi) [inline instat/SPEI.Designer.vb]: these too need to follow the right naming conventions. 

eg grpXXXXx

File: instat/SPEI.Designer.vb [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+        '
+        'UcrInputTextBox1
+        '
+        Me.UcrInputTextBox1.AddQuotesIfUnrecognised = True
+        Me.UcrInputTextBox1.IsMultiline = False
+        Me.UcrInputTextBox1.IsReadOnly = False
+        Me.UcrInputTextBox1.Location = New System.Drawing.Point(98, 360)
+        Me.UcrInputTextBox1.Name = "UcrInputTextBox1"
+        Me.UcrInputTextBox1.Size = New System.Drawing.Size(79, 31)
+        Me.UcrInputTextBox1.TabIndex = 18
```
Comment (maxwellfundi) [inline instat/SPEI.Designer.vb]: for ucr inputs- we name them like this - ucrInputXXXXXX

File: instat/SPEI.vb [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        UcrReceiverDate.AddIncludedMetadataProperty("Climatic_Type", {Chr(34) & "date" & Chr(34)})
+        UcrReceiverDate.bAutoFill = True
+        UcrReceiverDate.strSelectorHeading = "Date Variables"
+
+        ucrChkOmitMissingValues.SetParameter(New RParameter("na.rm", 6))
+        ucrChkOmitMissingValues.SetText("Omit Missing Values")
+        ucrChkOmitMissingValues.SetValuesCheckedAndUnchecked("TRUE", "FALSE")
+        ucrChkOmitMissingValues.SetRDefault("FALSE")
+
+    End Sub
```
Comment (shadrackkibet) [inline instat/SPEI.vb]: Always have one space between subs.

**Comment by shadrackkibet** [comment]
@thushari93 Thanks for sharing this code  congrats!.  However , there are conflicts which needs to be resolved. 
We always do frmMain changes separately so that they get merged quickly to avoid conflicts. Always ensure your branch is up to date with the master branch before making any changes.

@maxwellfundi  what are your suggestions here? should this be closed?

File: instat/SPEI.Designer.vb [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+        Me.rdoPET.Name = "rdoPET"
+        Me.rdoPET.Size = New System.Drawing.Size(38, 23)
+        Me.rdoPET.TabIndex = 3
+        Me.rdoPET.TabStop = True
+        Me.rdoPET.Text = "PET"
+        Me.rdoPET.UseVisualStyleBackColor = True
+        '
+        'LblDate
+        '
+        Me.LblDate.AutoSize = True
```
Comment (shadrackkibet) [inline instat/SPEI.Designer.vb]: always start with small l for labels e.g lblDate

File: instat/SPEI.Designer.vb [hunk 6]
```diff
@@ -0,0 +1,324 @@
...
+        Me.UcrReceiverYear.strNcFilePath = ""
+        Me.UcrReceiverYear.TabIndex = 6
+        Me.UcrReceiverYear.ucrSelector = Nothing
+        '
+        'UcrReceiverDate
+        '
+        Me.UcrReceiverDate.frmParent = Me
+        Me.UcrReceiverDate.Location = New System.Drawing.Point(261, 108)
+        Me.UcrReceiverDate.Margin = New System.Windows.Forms.Padding(0)
+        Me.UcrReceiverDate.Name = "UcrReceiverDate"
```
Comment (shadrackkibet) [inline instat/SPEI.Designer.vb]: start with small `u` i.e `ucrReceiverDate`

**Comment by maxwellfundi** [comment]
@shadrackkibet  I can try to manually remove the conflicts if its not a very complicated conflict. Else we will have to advice that he updates from the master then copy code back

File: instat/SPEI.vb [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+    Private Sub SetDefaults()
+        clsSPIFunction.SetPackageName("SPEI")
+        clsSPIFunction.SetRCommand("spei")
+        clsSPIFunction.AddParameter("data")
+        clsSPIFunction.AddParameter("scale")
+        clsSPIFunction.AddParameter("na.rm")
```
Comment (shadrackkibet) [inline instat/SPEI.vb]: I think you need parameter values and their positions here.

File: instat/SPEI.vb [hunk 8]
```diff
@@ -36,81 +37,91 @@ Public Class SPI
 
 
     Private Sub InitialiseDialog()
-        UcrBase.clsRsyntax.iCallType = 0
-        UcrBase.iHelpTopicID = 510
+        ucrBase.clsRsyntax.iCallType = 0
```
Comment (maxwellfundi) [inline instat/SPEI.vb]: if we need an output to be displayed in the output window, we need to haveiCallType = 2

**Pull request closed without merging**